### PR TITLE
is_empty for OsStr

### PR DESCRIPTION
**Pull request by ticki**
Part of #30520

### CODE REVIEW

**Comment by Ticki** [comment]
fixes #30259

**Comment by Gankro** [comment]
Travis seems unhappy

**Comment by Ticki** [comment]
It compiles on my machine, and the tests runs fine. I don't understand why travis is sad. I'm also unable to find travis's complains in the travis log.

**Comment by Gankro** [comment]
```

failures:

---- path::Path::is_empty_0 stdout ----
	<anon>:8:19: 8:29 error: use of unstable library feature 'os_extras': recently added (see issue #30259)
<anon>:8     assert!(!path.is_empty());
                           ^~~~~~~~~~
<anon>:8:5: 8:31 note: in this expansion of assert! (defined in <std macros>)
<anon>:8:19: 8:29 help: add #![feature(os_extras)] to the crate attributes to enable
error: aborting due to previous error
thread 'path::Path::is_empty_0' panicked at 'Box<Any>', src/libsyntax/errors/mod.rs:269


failures:
    path::Path::is_empty_0

test result: [31mFAILED(B[0m. 443 passed; 1 failed; 11 ignored; 0 measured
```

File: src/libstd/path.rs [hunk 1]
```diff
@@ -1859,6 +1860,23 @@ impl Path {
     pub fn is_dir(&self) -> bool {
         fs::metadata(self).map(|m| m.is_dir()).unwrap_or(false)
     }
+
+    /// Checks if the path buffer is empty. On Windows, it will return false if the inner string is
+    /// invalid unicode. On Unix, this is a no-op.
+    ///
+    /// # Examples
+    ///
+    /// ```
```
Comment (frewsxcv) [inline src/libstd/path.rs]: You need to add a `#![feature(os_extras)]` here in this code block

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to resubmit with tests passing!

**Comment by frewsxcv** [comment]
Rebased this and fixed in https://github.com/rust-lang/rust/pull/31231

**Comment by Ticki** [comment]
Thanks @frewsxcv!

**Pull request closed without merging**